### PR TITLE
[CIFuzz] Support non-github action CI for external projects

### PR DESCRIPTION
**Pull request by jonathanmetzman**
This is incomplete for a few reasons.
1. It probably can't diff properly.
2. It assumes the project source should be put in /src/$PROJECT_NAME.

### CODE REVIEW

File: infra/cifuzz/build_fuzzers.py [hunk 1]
```diff
@@ -72,6 +72,7 @@ def build_image_and_checkout_src(self):
       return False
     self.image_repo_path = result.image_repo_path
     self.repo_manager = result.repo_manager
+    logging.error('repo_dir %s', self.repo_manager.repo_dir)
```
Comment (inferno-chromium) [inline infra/cifuzz/build_fuzzers.py:75]: bad debug log statement?, why error

File: infra/cifuzz/continuous_integration.py [hunk 2]
```diff
@@ -191,6 +196,23 @@ def build_external_project_docker_image(project_name, project_src,
   return helper.docker_build(command)
 
 
+class ExternalGeneric(BaseCi):
+  def get_diff_base(self):
+    return 'origin...'
+
+  def prepare_for_fuzzer_build(self):
+    logging.info('YAAAAAAAY')
```
Comment (inferno-chromium) [inline infra/cifuzz/continuous_integration.py:204]: bad debug log statement?

File: infra/cifuzz/continuous_integration.py [hunk 3]
```diff
@@ -191,6 +196,23 @@ def build_external_project_docker_image(project_name, project_src,
   return helper.docker_build(command)
 
 
+class ExternalGeneric(BaseCi):
+  def get_diff_base(self):
```
Comment (inferno-chromium) [inline infra/cifuzz/continuous_integration.py:200]: not used anywhere?

File: infra/cifuzz/continuous_integration.py [hunk 4]
```diff
@@ -191,6 +196,23 @@ def build_external_project_docker_image(project_name, project_src,
...
+    return 'origin...'
+
+  def prepare_for_fuzzer_build(self):
+    logging.info('YAAAAAAAY')
+    manager = repo_manager.RepoManager(self.config.project_src_path)
+    abs_build_integration_path = os.path.join(manager.repo_dir,
+                                              self.config.build_integration_path)
+    if not build_external_project_docker_image(
+        self.config.project_name, manager.repo_dir, abs_build_integration_path):
+      logging.error('Failed to build external project.')
```
Comment (inferno-chromium) [inline infra/cifuzz/continuous_integration.py:210]: maybe add project name to message.

File: infra/cifuzz/continuous_integration.py [hunk 5]
```diff
@@ -191,6 +196,23 @@ def build_external_project_docker_image(project_name, project_src,
...
+      return BuildPreparationResult(False, None, None)
```
Comment (inferno-chromium) [inline infra/cifuzz/continuous_integration.py:211]: nit:named args is better for readability.
xyz=False, abc=None, def=None

File: infra/cifuzz/continuous_integration.py [hunk 6]
```diff
@@ -191,6 +196,23 @@ def build_external_project_docker_image(project_name, project_src,
...
 
 
+class ExternalGeneric(BaseCi):
+  def get_diff_base(self):
+    return 'origin...'
+
+  def prepare_for_fuzzer_build(self):
+    logging.info('YAAAAAAAY')
+    manager = repo_manager.RepoManager(self.config.project_src_path)
+    abs_build_integration_path = os.path.join(manager.repo_dir,
```
Comment (inferno-chromium) [inline infra/cifuzz/continuous_integration.py:206]: build_integration_abs_path reads better slightly

**Review by inferno-chromium — APPROVED** [review]

**Comment by jonathanmetzman** [comment]
CC @mpherman2

File: infra/cifuzz/build_fuzzers.py (hunk 1 above)
Comment (jonathanmetzman) [reply infra/cifuzz/build_fuzzers.py:75]: Good catch.

File: infra/cifuzz/continuous_integration.py (hunk 2 above)
Comment (jonathanmetzman) [reply infra/cifuzz/continuous_integration.py:204]: Good point.

File: infra/cifuzz/continuous_integration.py (hunk 3 above)
Comment (jonathanmetzman) [reply infra/cifuzz/continuous_integration.py:200]: It is used by the parent class in `get_changed_code_under_test`. This class inherits that function.

File: infra/cifuzz/continuous_integration.py (hunk 4 above)
Comment (jonathanmetzman) [reply infra/cifuzz/continuous_integration.py:210]: Done.

File: infra/cifuzz/continuous_integration.py (hunk 6 above)
Comment (jonathanmetzman) [reply infra/cifuzz/continuous_integration.py:206]: Done.

File: infra/cifuzz/continuous_integration.py (hunk 5 above)
Comment (jonathanmetzman) [reply infra/cifuzz/continuous_integration.py:211]: Done.

**Pull request merged**